### PR TITLE
Fix Android build

### PR DESCRIPTION
**Pull request by mcclure**
- Fix typos after "os" refactor
- Fix CMake finding javac

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -637,6 +637,7 @@ elseif(ANDROID)
   )
 
   if(LOVR_BUILD_EXE)
+    find_package(Java)
```
Comment (bjornbytes) [inline CMakeLists.txt:640]: This already exists on line 59, which confuses me

Comment (mcclure) [reply CMakeLists.txt:640]: Then I think this one is a merge error specific to my branch.

I will retest and remove this line.

File: src/core/os_android.c [hunk 2]
```diff
@@ -280,27 +280,27 @@ void os_poll_events() {
   }
 }
 
-void os_on_quit(quitCallback callback) {
+void os_on_quit(fn_quit callback) {
```
Comment (bjornbytes) [inline src/core/os_android.c:283]: Also I think the function pointers here should have a star after them e.g. `fn_quit*`

Comment (mcclure) [reply src/core/os_android.c:283]: You are right, as I wrote them these functions would crash if called

**Comment by mcclure** [comment]
I tested this new version with lovr-lodr. It addresses the issues above. I still was not able to get it to run without the `-0 so` -> `-0 ''` change we discussed in Slack.

**Comment by bjornbytes** [comment]
The `-0 so` thing can be tracked here https://github.com/nothings/stb/pull/960#discussion_r596584133

But I am surprised that `0bb556554dd59ca706290cfad4796f410631924c` didn't fix it

**Comment by mcclure** [comment]
I didn't test with 0bb5565. I tested with 6eef950aa84 (the parent of this patch)

**Comment by mcclure** [comment]
I pushed the wrong commit on Friday :( :(

Third try!! How's this?

**Pull request merged**